vulnerabilities/api.py

```diff
@@ -49,18 +49,17 @@ class MinimalPackageSerializer(serializers.HyperlinkedModelSerializer):
     affected_by_vulnerabilities = serializers.SerializerMethodField("get_affected_vulnerabilities")
 
     def get_affected_vulnerabilities(self, package):
-        parent_affected_vulnerabilities = package.fixed_package_details.get("vulnerabilities", None)
+        parent_affected_vulnerabilities = package.fixed_package_details.get("vulnerabilities") or []
         affected_vulnerabilities = []
 
-        if parent_affected_vulnerabilities:
-            for vuln in parent_affected_vulnerabilities:
-                affected_vulnerability = {}
+        for vuln in parent_affected_vulnerabilities:
+            affected_vulnerability = {}
 
-                affected_vulnerability["vulnerability"] = vuln.get(
-                    "vulnerability", None
-                ).vulnerability_id
+            affected_vulnerability["vulnerability"] = vuln.get(
```

Comment:
Thanks @TG1999.  I think I'm overlooking something obvious but . . .  I thought I was testing that with a case where the `vuln` has no `vulnerability` property using this test definition of the package:

```
searched_for_package = self.package_maven_jackson_databind_2_14_0_rc1
```

In the setup, I defined that package to have 0 vulnerabilities (as is the case in the DB as well), which would mean, I thought, that when we run that package through `MinimalPackageSerializer.get_affected_vulnerabilities()`, we are testing a `vuln` that has no `vulnerability` property.  

Thinking....

Reply:
@TG1999 Do I want to create a "fake" package structure where the `vulnerabilities` field has no `vulnerability` key but is not an empty list because it has at least one other key, e.g.,

```
        expected_searched_for_package_details = {
            "purl": PackageURL(
                type="maven",
                namespace="com.fasterxml.jackson.core",
                name="jackson-databind",
                version="2.14.0-rc1",
                qualifiers={},
                subpath=None,
            ),
            "next_non_vulnerable": None,
            "latest_non_vulnerable": None,
            "vulnerabilities": [{"foo": "bar"}],
        }
```

Reply:
@TG1999 While I was drafting this comment I made one more adjustment, will commit and push shortly, and then ready for your review next week.  

I've modified the `MinimalPackageSerializer()` code including a new `get_vulnerability()` class method and updated the related test `test_api_with_package_with_no_vulnerabilities()`.  When the `vuln` has no `vulnerability` key, e.g.,

```
        vuln = {
            "foo": "bar",
        }
```

`get_vulnerability()` adds no data to the `affected_vulnerabilities` list; and in the test, `MinimalPackageSerializer.get_vulnerability(self, vuln, affected_vulnerabilities)` is `None`.